Sources/PackageModel/Target.swift

```diff
@@ -445,9 +445,89 @@ public final class ClangTarget: Target {
 }
 
 public final class BinaryTarget: Target {
+    /// The kind of binary artifact.
+    public let kind: Kind
 
     /// The original source of the binary artifact.
-    public enum ArtifactSource: Equatable, Codable {
+    public let origin: Origin
+
+    /// The binary artifact path.
+    public var artifactPath: AbsolutePath {
+        return self.sources.root
+    }
+
+    public init(
+        name: String,
+        kind: Kind,
+        platforms: [SupportedPlatform] = [],
+        path: AbsolutePath,
+        origin: Origin
+    ) {
+        self.origin = origin
+        self.kind = kind
+        let sources = Sources(paths: [], root: path)
+        super.init(
+            name: name,
+            defaultLocalization: nil,
+            platforms: platforms,
+            type: .binary,
+            sources: sources,
+            dependencies: [],
+            buildSettings: .init()
+        )
+    }
+
+    private enum CodingKeys: String, CodingKey {
+        case kind
+        case origin
+        case artifactSource // backwards compatibility 2/2021
+    }
+
+    public override func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(self.origin, forKey: .origin)
+        try container.encode(self.kind, forKey: .kind)
+    }
+
+    required public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        // backwards compatibility 2/2021
+        if !container.contains(.kind) {
+            self.kind = .xcframework
+        } else {
+            self.kind = try container.decode(Kind.self, forKey: .kind)
+        }
+        // backwards compatibility 2/2021
+        if container.contains(.artifactSource)  {
+            self.origin = try container.decode(Origin.self, forKey: .artifactSource)
+        } else {
+            self.origin = try container.decode(Origin.self, forKey: .origin)
+        }
+        try super.init(from: decoder)
+    }
+
+    public enum Kind: String, RawRepresentable, Codable, CaseIterable {
+        case xcframework
+        // TODO: add new types
+        //case swiftLibraryArchive
+        //case swiftExecutableArchive
```

Comment:
I'm guessing that we're holding off on defining libraries for now (even in subsequent PRs), but it's good to have this marker here for the future.